### PR TITLE
Feat/marketplace

### PR DESCRIPTION
**Pull request by Mitsichury**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,15 +9,16 @@
     "registry": "node ./dist/registry.js",
     "compile": "npx tsc",
     "test": "jest --verbose --coverage --no-cache",
-    "bench": "node benchmarking/bench.js",
+    "benchERC20": "node benchmarking/benchERC20.js",
+    "benchMarketplace": "node benchmarking/benchMarketPlace.js",
```
Comment (dmosites) [inline package.json:13]: Let's make this just `bench` and `benchERC20`. In general, there is no marketplace protocol: Swap supports multiple token kinds, SwapERC20 supports ERC20 only.

File: src/client/Web3SwapClient.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            console.warn("Tried to add this network but it does not work :", network)
+            return;
+        }
+        if (this.keyExists(String(chainId))) {
+            console.log("Already connected");
+            return;
+        }
+
+        const provider = ethers.providers.InfuraProvider.getWebSocketProvider(chainId, this.apiKey);
+        const contract = Swap.getContract(provider, chainId);
```
[the remark this answers is not in the document]
Comment (piersss) [reply src/client/Web3SwapClient.ts:28]: Oh yeah I see there's an `deleteExpiredOrderERC20`. Maybe add it to the `Web3SwapClient.test.ts`?

[the remark this answers is not in the document]
Comment (Mitsichury) [reply src/client/Web3SwapClient.ts:28]: It's fixed 👍

[the remark this answers is not in the document]
Comment (Mitsichury) [reply src/client/Web3SwapClient.ts:28]: https://github.com/airswap/airswap-indexer/pull/36/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80

File: package.json [hunk 3]
```diff
@@ -9,15 +9,16 @@
...
   },
   "type": "module",
   "license": "MIT",
   "dependencies": {
-    "@airswap/libraries": "4.0.5-beta.5",
+    "@airswap/libraries": "4.0.5-beta.7",
     "@airswap/swap-erc20": "4.0.2",
-    "@airswap/types": "4.0.3-beta.0",
+    "@airswap/types": "4.0.3-beta.1",
     "@airswap/utils": "4.0.5-beta.1",
```
Comment (dmosites) [inline package.json:22]: Please upgrade to latest like so:
```
    "@airswap/libraries": "^4.0.0",
    "@airswap/types": "^4.0.0",
    "@airswap/utils": "^4.0.0",
```
`@airswap/swap-erc20` is included in `@airswap/libraries` so should not need to be added separately.

**Review by dmosites — APPROVED** [review]

**Pull request merged**